internal/alertmanager/currentalert/process.go

```diff
@@ -262,20 +263,26 @@ func resizePvc(log logr.Logger, labels model.LabelSet, annotiations model.LabelS
 
 			storageConfigs := brokerConfig.StorageConfigs
 
-			for n, c := range storageConfigs {
+			for _, c := range storageConfigs {
 				modifiableConfig := c.DeepCopy()
 				if modifiableConfig.MountPath == pvc.Annotations["mountPath"] {
 					size := *modifiableConfig.PvcSpec.Resources.Requests.Storage()
 					size.Add(incrementBy)
 
 					modifiableConfig.PvcSpec.Resources.Requests["storage"] = size
 
+					// When the storage is in brokerConfigGroup we don't resize the storage there because in that case
+					// all of those brokers that are using this brokerConfigGroup its storage would be resized.
+					// We add the storage into the broker.BrokerConfig.StorageConfigs in this way only that broker pvc will be resized.
```

Comment:
```suggestion
					// When the storage is in a brokerConfigGroup we don't resize the storage there because in that case
					// all of the brokers that are using this brokerConfigGroup would have their storages resized.
					// We add the storage into the broker.BrokerConfig.StorageConfigs in this way only the PVC belonging to that specific broker will be resized.
```

Reply:
Fixed in [`519a9c9`](https://github.com/banzaicloud/koperator/pull/958/commits/519a9c99e451bd25501c2194671bde41b1693f7d).